### PR TITLE
[PLA-1804] Events broadcast & logs refactor

### PR DESCRIPTION
**Pull request by leonardocustodio**
### **PR Type**
enhancement, bug fix

___

### **Description**
- Refactored event handling and broadcasting for `AuctionFinalized`, `BidPlaced`, `ListingCancelled`, `ListingCreated`, and `ListingFilled` events to use specific event classes and simplify data assignment.
- Added `log` and `broadcast` methods to event processing classes for better separation of concerns.
- Updated testbench configuration to include `SYNC_ALL` environment variable.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>10 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>AuctionFinalized.php</strong><dd><code>Refactor AuctionFinalized event handling and broadcasting</code></dd></summary>
<hr>

src/Events/Substrate/Marketplace/AuctionFinalized.php
<li>Refactored constructor to use <code>AuctionFinalizedPolkadart</code> event.<br> <li> Simplified broadcast data assignment using <code>toBroadcast</code> method.<br> <li> Updated broadcast channels to use event and extra data.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-marketplace/pull/46/files#diff-447301d68d138d7862b12ed6656273b0aa30e426afadabdd0a74c0e4ae10ca4e">+9/-25</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>BidPlaced.php</strong><dd><code>Refactor BidPlaced event handling and broadcasting</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Events/Substrate/Marketplace/BidPlaced.php
<li>Refactored constructor to use <code>BidPlacedPolkadart</code> event.<br> <li> Simplified broadcast data assignment using <code>toBroadcast</code> method.<br> <li> Updated broadcast channels to use event and extra data.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-marketplace/pull/46/files#diff-48d71f01225a2c70dd5137ead390642f90feb66674eb4d4d420062f345c36a19">+9/-23</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ListingCancelled.php</strong><dd><code>Refactor ListingCancelled event handling and broadcasting</code></dd></summary>
<hr>

src/Events/Substrate/Marketplace/ListingCancelled.php
<li>Refactored constructor to use <code>ListingCancelledPolkadart</code> event.<br> <li> Simplified broadcast data assignment using <code>toBroadcast</code> method.<br> <li> Updated broadcast channels to use event and extra data.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-marketplace/pull/46/files#diff-faa914304345724b0792afb22b63af259c19d823571a1253bc18e8d142a0f749">+8/-19</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ListingCreated.php</strong><dd><code>Refactor ListingCreated event handling and broadcasting</code>&nbsp; &nbsp; </dd></summary>
<hr>

src/Events/Substrate/Marketplace/ListingCreated.php
<li>Refactored constructor to use <code>ListingCreatedPolkadart</code> event.<br> <li> Simplified broadcast data assignment using <code>toBroadcast</code> method.<br> <li> Updated broadcast channels to use event and extra data.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-marketplace/pull/46/files#diff-1e2b23f48c3807e2fc90a2b5cbfa30185c0fff771adaf8b58bef836c2cfeda2c">+8/-31</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ListingFilled.php</strong><dd><code>Refactor ListingFilled event handling and broadcasting</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Events/Substrate/Marketplace/ListingFilled.php
<li>Refactored constructor to use <code>ListingFilledPolkadart</code> event.<br> <li> Simplified broadcast data assignment using <code>toBroadcast</code> method.<br> <li> Updated broadcast channels to use event and extra data.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-marketplace/pull/46/files#diff-90fcec9465855f220b0e3df55feb892d78f240f4de3204a79f0e373239b59a13">+9/-21</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>AuctionFinalized.php</strong><dd><code>Refactor AuctionFinalized event processing and logging</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Services/Processor/Substrate/Events/Implementations/Marketplace/AuctionFinalized.php
<li>Refactored <code>run</code> method to use class properties.<br> <li> Added <code>log</code> and <code>broadcast</code> methods.<br> <li> Simplified event handling and logging.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-marketplace/pull/46/files#diff-a59eaf43be85c9344f6d8d1b9f71fbdf806519510403c6f40712686df45019ee">+42/-33</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>BidPlaced.php</strong><dd><code>Refactor BidPlaced event processing and logging</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Services/Processor/Substrate/Events/Implementations/Marketplace/BidPlaced.php
<li>Refactored <code>run</code> method to use class properties.<br> <li> Added <code>log</code> and <code>broadcast</code> methods.<br> <li> Simplified event handling and logging.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-marketplace/pull/46/files#diff-87e8147ee2d3b1e5d4b9bf1d18ea5092955bd9aa72063f6f2ab7d223056317b8">+39/-30</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ListingCancelled.php</strong><dd><code>Refactor ListingCancelled event processing and logging</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Services/Processor/Substrate/Events/Implementations/Marketplace/ListingCancelled.php
<li>Refactored <code>run</code> method to use class properties.<br> <li> Added <code>log</code> and <code>broadcast</code> methods.<br> <li> Simplified event handling and logging.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-marketplace/pull/46/files#diff-6208097e959d26bcb08d7dc0b9abf5f62f9c842933ca08711eb8ef56e43e4f9d">+36/-26</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ListingCreated.php</strong><dd><code>Refactor ListingCreated event processing and logging</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Services/Processor/Substrate/Events/Implementations/Marketplace/ListingCreated.php
<li>Refactored <code>run</code> method to use class properties.<br> <li> Added <code>log</code> and <code>broadcast</code> methods.<br> <li> Simplified event handling and logging.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-marketplace/pull/46/files#diff-ce985e5f79fc66dd046576a93cbfa8ab99573a6a4c3920ad1acb62c103f9eb26">+44/-35</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ListingFilled.php</strong><dd><code>Refactor ListingFilled event processing and logging</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Services/Processor/Substrate/Events/Implementations/Marketplace/ListingFilled.php
<li>Refactored <code>run</code> method to use class properties.<br> <li> Added <code>log</code> and <code>broadcast</code> methods.<br> <li> Simplified event handling and logging.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-marketplace/pull/46/files#diff-39ed01e1015d29205b4058399227d1051514d5c1808c1d0b93e23c143b40b654">+39/-30</a>&nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>testbench.yaml</strong><dd><code>Add SYNC_ALL environment variable to testbench configuration</code></dd></summary>
<hr>

testbench.yaml
- Added `SYNC_ALL` environment variable.

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-marketplace/pull/46/files#diff-ef7834cd1991a169c5e09ee7b61b72255e1feabf34bb85a21c75fc6fa532917a">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: src/Events/Substrate/Marketplace/AuctionFinalized.php [hunk 1]
```diff
@@ -6,43 +6,27 @@
...
 {
     /**
      * Create a new event instance.
      */
-    public function __construct(Model $listing, Model $state, Model $sale, ?Model $transaction = null)
+    public function __construct(AuctionFinalizedPolkadart $event, ?Model $transaction = null, ?array $extra = null)
     {
         parent::__construct();
 
-        $this->model = $sale;
```
Comment (enjinabner) [inline src/Events/Substrate/Marketplace/AuctionFinalized.php:19]: Did we remove this for platform-core as well? This is attribute being used in https://github.com/enjin/platform-multi-tenant/blob/master/src/Listeners/LinkEventDataToTenant.php#L24

**Review by enjinabner — CHANGES_REQUESTED** [review]
Clarification

File: src/Events/Substrate/Marketplace/AuctionFinalized.php (hunk 1 above)
Comment (enjinabner) [reply src/Events/Substrate/Marketplace/AuctionFinalized.php:19]: Here as well https://github.com/enjin/platform-multi-tenant/blob/master/src/Listeners/LinkRelayTransactionToTenant.php#L18

Comment (leonardocustodio) [reply src/Events/Substrate/Marketplace/AuctionFinalized.php:19]: Yep, it was removed from there. I also asked here: https://enjin.slack.com/archives/C04LN0G5F2P/p1717166162432909
I can check what I can do to put it back, no problem.

**Review by enjinabner — APPROVED** [review]

**Pull request merged**